src/Dropdown/index.jsx

```diff
@@ -1,203 +1,125 @@
 import React from 'react';
-import classNames from 'classnames';
 import PropTypes from 'prop-types';
-
 import Button from '../Button';
 
-export const triggerKeys = {
-  OPEN_MENU: ['ArrowDown'],
-  CLOSE_MENU: ['Escape'],
-  NAVIGATE_DOWN: ['ArrowDown', 'Tab'],
-  NAVIGATE_UP: ['ArrowUp'],
-  SELECT_MENU_ITEM: ['Enter', ' '],
-};
-
 class Dropdown extends React.Component {
-  static isTriggerKey(action, keyName) {
-    return triggerKeys[action].indexOf(keyName) > -1;
-  }
-
   constructor(props) {
     super(props);
-
-    this.addEvents = this.addEvents.bind(this);
-    this.handleDocumentClick = this.handleDocumentClick.bind(this);
-    this.handleToggleKeyDown = this.handleToggleKeyDown.bind(this);
-    this.handleMenuKeyDown = this.handleMenuKeyDown.bind(this);
-    this.removeEvents = this.removeEvents.bind(this);
-    this.toggle = this.toggle.bind(this);
-
-    this.menuItems = [];
     this.state = {
       open: false,
-      focusIndex: 0,
     };
+    this.dropdownItemRef = React.createRef();
   }
 
-  componentWillUpdate(_, nextState) {
-    if (nextState.open) {
-      this.addEvents();
-    } else {
-      this.removeEvents();
-    }
+  componentDidMount = () => {
+    document.addEventListener('click', this.handleDocumentClick, true);
```

Comment:
We should only listen to click events on the document when we need to. Ideally we add a listener when the menu opens, remove the listener when it closes, and remove it when the component unmounts (just in case).

Reply:
Removed event listener from componentDidMount to open method.